core/src/main/scala/scala/pickling/spi/PicklerRegistry.scala

```diff
@@ -56,23 +56,22 @@ trait PicklerRegistry {
     *             In those situations, the pickler should be able to handle arbitrary (existential) type parameters.
     * @param p  The pickler to register.
     */
-  def registerPickler[T](key: String, p: Pickler[T]): Unit
-  def registerPickler[T](p: Pickler[T]): Unit = registerPickler(p.tag.key, p)
+  def registerPickler(key: String, p: Pickler[_]): Unit
+  def registerPickler(p: Pickler[_]): Unit = registerPickler(p.tag.key, p)
```

Comment:
Yeah, it may infer Any, or it may leave the type as an existential.

In any case, the errors we generally have are of the form of existential type errors, such as:

`cannot prove that _$21 is _$22` or some such nonesense.   Could be we're not using them heavily here, but I've just run into so many existential bugs, I'd prefer to leave the type parameters.  You're not gaining anything by dropping the type parameter.

That aside, it's kind of a useless argument here if the code works both ways. I won't force it, but I will adamantly encourage you to prefer type parameters over existentials in general.   Type inference tends to work better, and we'll have less issues if we do write abstract code over these methods.   

In sbt, we have code that works over `Initialize[_]` and `Initialize[Task[_]]`, in tandem with `SettingKey[_]` and `TaskKey[_]` respectively.    It's similar to here where `FastTypeTag[T]` is meant to line up with `Pickler[T]`.   We eventually refactoring most existential code to just be type parameters because the number of instances where the existential is a pest and requires explicit typing is much higher than using type paramters.


Think of it this way.   Existentials => You have to prove one type exists everywhere.   Type parameter => You prove the type exists at the call site and then you can freely use the parameter all the way down with little inference penalty.

This comes into play mostly with dependent functions, so if you write something like:

`def doStuffWithKey[T](key: FastTypeTag[T], p: Pickler[T]) = ....`

It'll go smoother if you lift the T as soon as you can, vs. trying to get by with:

`def doStuffWithKey(key: FastTypeTag[_], p: Pickler[_]) = ....`

Those two functions are VERY different in usage.  I can't remember all my old tricks, but IIRC we did have one to unify the existential on the map of paired Key[T]/Value[T] items, e.g.
```scala
def handleItem[A](key: Key[A], value: Value[A]): Result = 
   ... here we use key/value freely and know they line up....

// Here you hackily force the type unification of your existentials. You *only* want to do this when you're sure it's safe, because this is literally a "soundness hole" you've opened in your program.
for ( (key, value) <- keyValueMap) 
yield handleItem[Any](key, value)
```